src/qtcore/qml/qml.js

```diff
@@ -209,23 +209,32 @@ function construct(meta) {
     return item;
 }
 
+const typeDefaultValues = {
+  int: 0,
+  real: 0,
+  double: 0,
+  string: '',
+  bool: false,
+  list: [],
+  url: ''
+};
+
 /**
  * Create property getters and setters for object.
  * @param {Object} obj Object for which gsetters will be set
  * @param {String} propName Property name
  * @param {Object} [options] Options that allow finetuning of the property
  */
-function createSimpleProperty(type, obj, propName, options) {
+function createProperty(type, obj, propName, options = {}) {
     var prop = new QMLProperty(type, obj, propName);
     var getter, setter;
 
-    if (typeof options == 'undefined')
-      options = {};
-    else if (typeof options != 'object')
-      options = { default: options }
-
     obj[propName + "Changed"] = prop.changed;
     obj.$properties[propName] = prop;
+    if (!options.hasOwnProperty('default')) {
```

Comment:
I think we should use `constructors[prop.type]();` (default constructor) for setting the default value. Either using `obj.$properties[propName].val = constructors[prop.type]();` or just using `obj.$properties[propName].set(options.default)` (if it's not set `options.default` is undefined, thus the constructor would return a default constructed variable. (In any case, we'd need to make sure that the constructor does that)